src/sonic-yang-models/yang-models/sonic-tunnel.yang

```diff
@@ -0,0 +1,117 @@
+module sonic-tunnel {
+    yang-version 1.1;
+    namespace "http://github.com/Azure/sonic-tunnel";
+    prefix tunnel;
+
+    import ietf-inet-types {
+        prefix inet;
+    }
+    
+    import sonic-types {
+        prefix stypes;
+    }
+
+    organization
+        "SONiC";
+
+    contact
+        "SONiC";
+
+    description
+        "SONiC DualToR tunnel data";
+
+    revision 2022-08-23 {
+        description
+            "Initial revision";
+    }
+
+    container sonic-tunnel {
+        container TUNNEL {
+            list TUNNEL_LIST {
+                max-elements 1;
+
+                key "mux_tunnel";
+                
+                leaf mux_tunnel {
+                    description "Name of MuxTunnel";
+                    type string {
+                        pattern "MuxTunnel[0-9]+";
+                    }
+                }
+
+                leaf dscp_mode {
+                    description "DSCP mode";
+                    type string {
+                        pattern "uniform|pipe";
+                    }
+                }
+
+                leaf src_ip {
+                    type inet:ipv4-address;
+                    description "optional source IPv4 address.";
```

Comment:
dst_ip is needed for the tunnel configuration, but src_ip was only used when tunnel_qos_remap = enabled. I'll specify that instead of saying it's optional